### PR TITLE
feat: adds hoc for making props consistent cross platform

### PR DESCRIPTION
**Pull request by dickie81**
## Description:
Adds HOC for making props passed from native to App consistent.

### Checklist for this pull request

* [x] PR is scoped to one task
* [ ] Tests are included
* [x] Documentation included
* [x] One of:
  * [x] The PR is not making any UI change
  * [ ] The PR is making a UI change but not a product change
    * [ ] Screenshots included

Thank you!

### CODE REVIEW

File: src/withZapp/index.js [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+const propsToParse = ['localization', 'settings', 'styles', 'extra_props'];
+
+// withZap is a react native HOC to patch over platform differences between iOS & android
+export default WrappedComponent =>
+  Platform.select({
+    ios: () => WrappedComponent,
+    android: () => {
+      const ZappedComponent = props => (
+        <WrappedComponent
+          {...Object.assign(
```
Comment (Budaa) [inline src/withZapp/index.js:13]: There is no need for Object.assign as spread operator is doing the same thing.

**Review by Budaa — CHANGES_REQUESTED** [review]
Good effort, I dropped a small ES6 related comment.

File: src/withZapp/index.js (hunk 1 above)
Comment (dickie81) [reply src/withZapp/index.js]: no they aint

Comment (Budaa) [reply src/withZapp/index.js:13]: After @dickie81 in-person intervention, I'm happily changing my mind as he was right. You need `Object.assign`

**Review by Budaa — APPROVED** [review]

**Pull request merged**